src/App.js

```diff
@@ -9,30 +9,48 @@ import CalculationState from "./context/calculation/CalculationState";
 
 // import './App.css';
 
-const App = () => {
-  return (
-    <TransactionState>
-      <CalculationState>
-        <Fragment>
-          <h1 className="center-align">Crypto Capital Gains Calculator</h1>
-          <div className="container">
-            <p className="flow-text center">
-              This app was built mainly to get practice designing and building
-              React apps. I am not an accountant and make no guarantees that the
-              calculations performed within this app are accurate. I encourage
-              you to seek the help of an accountant or tax expert with any
-              questions regarding tax obligations for crypto transactions.
-            </p>
-          </div>
-          <div className="divider"></div>
-          <TransactionForm />
-          <TransactionTable />
-          <ErrorMessage />
-          <CalculationTable />
-        </Fragment>
-      </CalculationState>
-    </TransactionState>
-  );
-};
+class App extends Component {
+  state = {
+    errorState: {
+      dateOfFirstInvalidTransaction: null,
+    },
+  };
+
+  setDateOfFirstInvalidTransaction = (date) =>
```

Comment:
i think it would be fair to call this `showError(mesage)` in this pr, and update the transaction stuff to create the message itself

Reply:
Updated all components that referenced date of first invalid transaction to now reference an error message. Refactored the ErrorMessage component to accept the full message instead of just the date. simplified the state for the App component so that it is no longer an errorState object and is just errorMessage.